lib/spack/spack/spec.py

```diff
@@ -1662,59 +1663,88 @@ def to_node_dict(self, hash=ht.dag_hash):
                 package_hash = package_hash.decode('utf-8')
             d['package_hash'] = package_hash
 
+        # Note: Depends on key sort.
```

Comment:
This should comment should be more descriptive. Maybe "Note: relies on sorting dict by keys later in algorithm" or something of the sort.